src/vmm/src/lib.rs

```diff
@@ -421,20 +422,38 @@ impl Vmm {
         self.vm.register_irqfd(&interrupt_evt, 4)?;
 
         self.kernel_cfg.cmdline.push_str(" console=ttyS0");
+        #[cfg(target_arch = "aarch64")]
+        self.kernel_cfg
+            .cmdline
+            .push_str(&format!(" earlycon=uart,mmio,0x{:08x}", AARCH64_MMIO_BASE));
+
         // Put it on the bus.
-        // Safe to use expect() because the device manager is instantiated in new(), there's no
+        // Safe to use unwrap() because the device manager is instantiated in new(), there's no
         // default implementation, and the field is private inside the VMM struct.
-        self.device_mgr
-            .lock()
-            .unwrap()
-            .register_pio_resources(
-                serial.clone(),
-                &[Resource::PioAddressRange {
-                    base: 0x3f8,
-                    size: 0x8,
-                }],
-            )
-            .unwrap();
+        #[cfg(target_arch = "x86_64")]
+        {
+            self.device_mgr
+                .lock()
+                .unwrap()
+                .register_pio_resources(
```

Comment:
Can we use `register_pio()` here if we are registering only one resource?

Reply:
Yes, used `register_pio()` instead.